core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreMessageHandler.java

```diff
@@ -84,17 +118,85 @@ public void handleUpdate(SchemaRegistryKey key,
                            TopicPartition tp,
                            long offset,
                            long timestamp) {
+    if (key.getKeyType() != SchemaRegistryKeyType.SCHEMA && value == null) {
+      // ignore non-schema tombstone
+      return;
+    }
+
     if (key.getKeyType() == SchemaRegistryKeyType.SCHEMA) {
       handleSchemaUpdate((SchemaKey) key,
           (SchemaValue) value,
           (SchemaValue) oldValue);
-    } else if (value == null) {
-      // ignore non-schema tombstone
     } else if (key.getKeyType() == SchemaRegistryKeyType.DELETE_SUBJECT) {
       handleDeleteSubject((DeleteSubjectValue) value);
     } else if (key.getKeyType() == SchemaRegistryKeyType.CLEAR_SUBJECT) {
       handleClearSubject((ClearSubjectValue) value);
     }
+    recordBackup(key, value, tp, timestamp);
```

Comment:
should we call it only when backup is enabled?

Reply:
you're right, the check should be here